### PR TITLE
api:update:  publish and setdefault flags added with tests

### PR DESCRIPTION
**Pull request by ali-smartbear**
Adds feature publish and setdefault flags to update command

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,6 +202,9 @@ DESCRIPTION
 EXAMPLES
   swaggerhub api:update organization/api --file api.yaml
   swaggerhub api:update organization/api/1.0.0 --file api.json
+  swaggerhub api:update organization/api/1.0.0 --publish --file api.json
+  swaggerhub api:update organization/api/1.0.0 --setdefault --file api.json
+  swaggerhub api:update organization/api/1.0.0 --publish --setdefault --file api.jso
```
Comment (ponelat) [inline README.md:207]: typo: `api.jso` => `api.json`

File: src/commands/api/update.js [hunk 2]
```diff
@@ -64,6 +75,18 @@ UpdateAPICommand.flags = {
...
     default: 'private'
   }),
+  publish: flags.boolean({
+    description: 'publishes API when API gets updated',
+    default: false,
+    required: false,
+    dependsOn: ['file']
+  }),
+  setdefault: flags.boolean({
+    description: 'sets default API version to the version in the provided file when API gets updated',
```
Comment (ponelat) [inline src/commands/api/update.js:85]: `sets API version to be the default`. @hkosova thoughts?

File: test/commands/api/update.test.js [hunk 3]
```diff
@@ -77,6 +77,62 @@ describe('invalid api:update', () => {
...
+      .get('/org/api/1.0.0')
+      .reply(200)
+    )
+    .nock(`${shubUrl}/apis`, api => api
+      .post('/org/api?version=1.0.0&isPrivate=true')
+      .reply(200)
+    )
+    .nock(`${shubUrl}/apis`, api => api
+      .put('/org/api/1.0.0/settings/lifecycle')
+      .reply(404, '{ "code": 404, "message": "Unknown API org/api:1.0.0"}')
```
Comment (ponelat) [inline test/commands/api/update.test.js:105]: The mock error message isn't related to the test. The API is known, but it failed to be published.

File: test/commands/api/update.test.js [hunk 4]
```diff
@@ -77,6 +77,62 @@ describe('invalid api:update', () => {
...
+      .put('/org/api/1.0.0/settings/lifecycle', { published: true })
+      .reply(200)
+    )
+    .nock(`${shubUrl}/apis`, api => api
+      .put('/org/api/settings/default', { version: '1.0.0' })
+      .reply(404, '{ "code": 404, "message": "Unknown API org/api:1.0.0"}')
+    )
+    .command(['api:update', `${validIdentifier}`, '-f=test/resources/valid_api.yaml', '--publish', '--setdefault'])
+    .catch(err => {
+      expect(err.message).to.contains('Unknown API org/api:1.0.0')
```
Comment (ponelat) [inline test/commands/api/update.test.js:133]: Similar to previous note, this error message doesn't relate to the test. Please change wording.

File: src/commands/api/update.js [hunk 5]
```diff
@@ -64,6 +75,18 @@ UpdateAPICommand.flags = {
     options: ['public', 'private'],
     default: 'private'
   }),
+  publish: flags.boolean({
+    description: 'publishes API when API gets updated',
```
Comment (ponelat) [inline src/commands/api/update.js:79]: `sets the API version as published`.

**Review by ponelat — CHANGES_REQUESTED** [review]
Minor changes
- Wording of help text
- Update error messages in tests 
- Minor typo in readme

**Review by ponelat — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -192,6 +192,8 @@ OPTIONS
   -f, --file=file              (required) file location of API to update
   -h, --help                   show CLI help
   --visibility=public|private  [default: private] visibility of API in SwaggerHub
+  --publish                    publish the API version when it gets updated
```
Comment (conor-ward) [inline README.md:195]: I think this needs to be updated again to align with the description in the class

**Comment by conor-ward** [comment]
I got the following output after running with publish and setdefault
```
Updated API 'IntegrationsDemo/DemoAPI/1.0.4'
Published API IntegrationsDemo/DemoAPI/1.0.4
Default version of IntegrationsDemo/DemoAPI set to 1.0.4
```
It might be better to remove the quotes around the API version for updated to keep it consistent

**Review by conor-ward — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (barryadk) [reply README.md]: Probably don't need examples of every combination of flags. I think just the last example is necessary to show the publish and setdefault flags.
i.e. The following lines could be removed:
``` 
swaggerhub api:update organization/api/1.0.0 --publish --file api.json
swaggerhub api:update organization/api/1.0.0 --setdefault --file api.json

Comment (ali-smartbear) [reply README.md]: There is no harm if we keep them all. publish and setdefault are requiring file flag and it's good to show it in examples.

**Review by barryadk — APPROVED** [review]

**Pull request merged**